### ISSUE TITLE
errors in average aggregation of calculated field

### ISSUE THREAD

**Issue by yehosef** [elastic/elasticsearch#11476]
We were playing around with scripted fields in kibana and took a certain field (page number) and just set the scripted field to the page number *2.  It seemed to work fine but then on close inspection we noticed that sometimes it is off.

Below shows the chart of the percentage - this should always be 66% but sometimes its not.

If we run the same aggregation using the sum instead of the average, it is correct.

I haven't tried this in pure elasticsearch but it doesn't seem it should be a problem in kibana.  It's just taking the results from ES and graphing them AFAIK.

![visualize_-_kibana_4](https://cloud.githubusercontent.com/assets/149253/7962177/b229d134-0a14-11e5-8fb5-0f684cc5b83e.png)

![visualize_-_kibana_4](https://cloud.githubusercontent.com/assets/149253/7962211/da79784c-0a14-11e5-89bd-56e9a2759441.png)

**Comment by spinscale** [comment]
hey there,

a couple of questions here:

* Does this also happen, when you dont run it on a scripted field?
* Can you share the queries, that kibana executes (kibana allows you to show the full query)?
* Most importantly, do you have the chance to provide us the data which causes this problem? Can you provide a snapshot if the data is not too sensitive or help us creating a dataset that reproduces the problem?

**Comment by clintongormley** [comment]
Also wondering if some documents have multiple values in one field?

**Comment by yehosef** [comment]
@spinscale 
1) - I don't know if it's happens on a non-scripted field - I don't have any fields that are always the same ratio of another field for comparison
2)  https://gist.github.com/yehosef/f96dc491bcd5ee9bf7d3 is the query
I also included the index template - perhaps it's related to doc_values.  
3) I'm not sure if I can share the data right now - if you contact me at my name at gmail, I can probably give you temp access to the marvel/kibana install. 

To give some background - this is a testing server to test load/capabilities for a server.  We have storing all fields to "off" and are using doc_values.  It's a single 16G machine with 8G heap - 220M rows, ~60G storage.

**Comment by yehosef** [comment]
@clintongormley  - I'm pretty sure that it doesn't.  We're importing Mixpanel data and you don't have arrays in that data set.  We have found chunks of our import data that was truncated - but I think that would result in broken json and just wouldn't import those lines.

**Comment by clintongormley** [comment]
@yehosef broken JSON can well end up with bad results.  Our JSON parser is too lenient (fixed in 2.0) so only part of the JSON document might have been indexed.  It'd be worth comparing doc counts for the buckets with funny averages with buckets with good averages.

**Comment by yehosef** [comment]
It's hard for me to think that's the problem. There would have been one broken record per day (the last one) and even if it's broken how could that affect a computed field?  Whatever bad value there was would have been doubled.  And how would it only affect avg but not sum?  

But I'll investigated further and check with other data sources.

**Comment by clintongormley** [comment]
OK. I've indexed 100,000 documents with a timestamp and a number, and repeated your experiment.  The avg and sum lines are exactly equal. Then I added 10,000 documents with just a timestamp.  The sum line remains flat and the avg line shows differences.  So the difference that you are seeing is a result of your broken JSON, where some documents are missing values for this field (but the doc count is still taken into account when calculating the average).

This is not a bug.
